packages/grid/theme/lumo/vaadin-grid-styles.js

```diff
@@ -372,7 +372,7 @@ registerStyles(
       border-left: var(--_lumo-grid-border-width) solid transparent;
     }
 
-    :host([dir='rtl'][overflow~='right']) [part~='cell'][last-frozen]:not([part~='details-cell']) {
+    :host([dir='rtl'][overflow~='start']) [part~='cell'][last-frozen]:not([part~='details-cell']) {
```

Comment:
One day we might be able to simplify this by using only `border-inline-end-color` instead.